### DISCUSSION TITLE
Statistics about model simulations, e.g. run time, memory usage? And which compilers

### DISCUSSION THREAD

**Discussion by QinggangGao** [esm-tools/esm_tools#700] [Models]
Dear developers

I am developing ECHAM6 and might run awi-esm2.1 later. Taking these models as examples, may I ask several questions:

1) Is there a smarter and more concise way to check how long a model runs for one period, besides looking at log file?

2) And how to check how much memory it uses?

3) How to check which compiler does it use? e.g. which gfortran?

Regards, Qinggang

**Comment by QinggangGao** [comment]
And maybe another two questions:

4. In ECHAM6 with T63 resolution, delta_time is 450s, and time_step_len is 900s. I thought the timestep of ECHAM6 is 450s, so why time_step_len is also used, e.g. to estimate atmospheric humidity in the current time step from the value in the previous time step?

5. When I run simulations with setting `putdata: [ 450, 'seconds', 'first', 0 ]`, why the fist timestep is in 2000-01-01 00:15:00, rather than 00:07:00?

I will also search for answers in the user guide. Thanks in advance if you might have any ideas on it.

Regards, Qinggang

**Comment by mandresm** [comment]
1. not possible yet, but we'll keep that idea in mind (see #832)
2. not possible yet with ESM-Tools. YOu might be able to do that through slurm commands
3. in your folder where the model is installed, you'll find a `comp-<model_name_version>.sh` file. That's the script that is ultimately used for compilation and you can see there which modules are loaded and what's the compilation command.
4. Is this in the namelists or in the yamls?
5. I don't know, I think this is a question for an ECHAM expert. There is the need for a more extended description of the problem (namelists used, restart state, ...)
